### PR TITLE
Immuno rnaseq fusion enhancements

### PR DESCRIPTION
**Pull request by johnegarza**
This PR aims to enhance the immuno workflow's fusion calling ability and downstream analysis capability by adding agfusion and pvacfuse, as described in https://github.com/griffithlab/immuno-planning/issues/2. This also involved replacing the older rnaseq-hisat pipeline with the newer rnaseq-star + star-fusion pipeline within the immuno workflow. Several existing inputs in the rnaseq-star pipeline were renamed for clarity, and a separate read group input (`outsam_attrrg_line`) was dropped in favor of including it alongside the raw data files in the custom sequence type input. However, since the format for this read group string is different than the one expected within the sequence type input to the somatic/germline pipelines, docs have been added to both that describe the respective formats and provide examples.

### CODE REVIEW

File: definitions/tools/agfusion.cwl [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env cwl-runner
+
+class: CommandLineTool
+cwlVersion: "v1.0"
+label: "A tool that annotates STAR gene fusion predictions"
+
+baseCommand: ["/usr/local/bin/agfusion", "batch"]
+requirements:
+    - class: ShellCommandRequirement
```
Comment (tmooney) [inline definitions/tools/agfusion.cwl:9]: Is this using the `ShellCommandRequirement`?

File: definitions/tools/agfusion.cwl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        position: 3
+
+  output_dir:
+    type: string?
+    default: "agfusion_results"
+    inputBinding:
+        prefix: '-o'
+        position: 4
+
+
```
Comment (tmooney) [inline definitions/tools/agfusion.cwl:47]: (There's a lot of whitespace in this file compared to most of these.)

**Review by tmooney — COMMENTED** [review]
This generally looks okay to me!  I'm not familiar with `agfusion`, so I'm mostly trusting that it does what you want.

**Review by tmooney — APPROVED** [review]
+1 ⛽

**Review by jasonwalker80 — DISMISSED** [review]
+1

**Comment by gschang** [comment]
(+1)

**Review by susannasiebert — APPROVED** [review]
+1

**Comment by johnegarza** [comment]
Manually reference https://github.com/griffithlab/immuno-planning/issues/10 in order to reference the write-up as needed

**Pull request merged**